mautic/mautic
# Salesforce avoid duplicate contacts - new SF formula field matching

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y (enhancement)
| New feature? | y 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
With this PR the Salesforce fetch leads command will not import contacts that do not have any data for an unique field (eg: email) to avoid duplicating records. Also importing formula fields from SalesForce. 

#### Steps to test this PR:
1. Apply this PR
2. When running the fetch leads command (`php app/console mautic:integration:fetchleads --integration=Salesforce --time-interval="29 days"`) try importing leads that don't have email set for example if email is set as your unique identifier field. Mautic will not create a lead in this case since it is not able to identify the lead, it will duplicate it.
3. On the plugin configuration window for SalesForce, match a formula field with a field in Mautic, this should now be imported. Although it will not be pushed to Salesforce since formula fields are not updatable.
